test-case/check-pause-release-suspend-resume.sh

```diff
@@ -107,29 +99,34 @@ esac
 
 [[ -z $file_name ]] && file_name=$dummy_file
 
-
 logger_disabled || func_lib_start_log_collect
 
-setup_kernel_check_point
-
-dlogi "Entering audio stream expect script with: $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q"
-dlogi "Will enter suspend-resume cycle during paused period of audio stream process"
-
-rm -rf /tmp/sof-test.lock
-
-# expect is tcl language script
-#   catch: Evaluate script and trap exceptional returns
-#   after ms: Ms must be an integer giving a time in milliseconds.
-#       The command sleeps for ms milliseconds and then returns.
-expect <<AUDIO
-spawn $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q
+func_pipeline_export "$tplg" "type:$test_mode & ${OPT_VAL['p']}"
+
+for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+do
+    channel=$(func_pipeline_parse_value "$idx" channel)
+    rate=$(func_pipeline_parse_value "$idx" rate)
+    fmt=$(func_pipeline_parse_value "$idx" fmt)
+    dev=$(func_pipeline_parse_value "$idx" dev)
+    snd=$(func_pipeline_parse_value "$idx" snd)
```

Comment:
Before the user could pass these on the command line. Not anymore, isn't that also removing a feature? Please explain in the commit message.

Reply:
I don't remove a feature, now all these parameters can be parsed from the specified or default tplg file automatically. I explained in the commit message "This patch allows user to specify an input tplg file. Then the test case will call func_pipeline_parse_value to automatically obtain pipeline parameters from the tplg file.

User no longer needs to use parameter to specify each pipeline paremeters."

Reply:
OK the user can edit the topology file if needed, I did not consider that. Thanks for the explanation.